spyder/config/base.py

```diff
@@ -122,13 +122,18 @@ def get_home_dir():
 
 
 def get_conf_path(filename=None):
-    """Return absolute path for configuration file with specified filename"""
+    """Return absolute path to the config file with the specified filename."""
     # Define conf_dir
-    if running_under_pytest():
-        import py
-        from _pytest.tmpdir import get_user
-        conf_dir = osp.join(str(py.path.local.get_temproot()),
-                            'pytest-of-{}'.format(get_user()),
+    if running_under_pytest() or CLEAN:
+        # Use clean config dir if running tests or the user requests it.
+        import getpass  # analysis:ignore
+        if sys.platform.startswith("win"):
+            current_user = ''
+        else:
+            current_user = '-' + str(getpass.getuser())
+        import tempfile  # analysis:ignore
+        conf_dir = osp.join(str(tempfile.gettempdir()),
+                            'pytest-spyder{0!s}'.format(current_user),

```

Comment:
Let's move this whole block to a new function called `clean_conf_dir`

Reply:
Will do, thanks.

Reply:
Done; moved to ``get_clean_config_dir``.